src/Parser/Shaman/Elemental/Modules/ShamanCore/FrostShock.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/SPELLS/TALENTS/SHAMAN';
+import SpellLink from 'common/SpellLink';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+class FrostShock extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  badFrostShockCount = 0;
+  suboptimalFrostShockCount = 0;  //not used yet, wanna talk to tc-people for thresholds
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.FROST_SHOCK.id) {
```

Comment:
It's preferred to avoid nested if statements when possible. Instead check for the opposite and return early.